service/src/test/java/uk/gov/hmcts/reform/fpl/service/PrepareDirectionsForDataStoreServiceTest.java

```diff
@@ -0,0 +1,627 @@
+            return caseData.getServedCaseManagementOrders().get(0).getValue()
+                .getDirections().get(0).getValue().getResponses();
+        }
+    }
+
+    @Nested
+    class AddResponseElementsToDirection {
+        UUID responseId = randomUUID();
+        UUID directionId = randomUUID();
+
+        @Test
+        void shouldAddResponseElementWhenThereAreNoResponses() {
+            List<Element<DirectionResponse>> responses = createDirectionResponses(responseId, directionId);
+
+            List<Element<Direction>> directions = createDirections(directionId, new ArrayList<>());
+
+            service.addResponseElementsToDirections(responses, directions);
+
+            assertThat(directions.get(0).getValue().getResponses()).hasSize(1);
+        }
+
+        @Test
+        void shouldAddResponseElementWhenThereAreResponsesWithDifferentResponseId() {
+            List<Element<DirectionResponse>> responses = createDirectionResponses(responseId, directionId);
+
+            List<Element<Direction>> directions =
+                createDirections(directionId, createDirectionResponses(randomUUID(), directionId));
+
+            service.addResponseElementsToDirections(responses, directions);
+
+            assertThat(directions.get(0).getValue().getResponses()).hasSize(2);
+        }
+
+        @Test
+        void shouldReplaceResponseElementWhenThereIsResponsesWithSameResponseId() {
+            List<Element<DirectionResponse>> responses = createDirectionResponses(responseId, directionId);
+
+            List<Element<Direction>> directions =
+                createDirections(directionId, createDirectionResponses(responseId, directionId));
+
+            service.addResponseElementsToDirections(responses, directions);
+
+            assertThat(directions.get(0).getValue().getResponses()).hasSize(1);
+        }
+
+        @Test
+        void shouldNotAddResponseElementWhenDifferentDirectionId() {
+            List<Element<DirectionResponse>> responses = createDirectionResponses(responseId, randomUUID());
+
+            List<Element<Direction>> directions = createDirections(directionId, new ArrayList<>());
+
+            service.addResponseElementsToDirections(responses, directions);
+
+            assertThat(directions.get(0).getValue().getResponses()).isEmpty();
+        }
+
+        private List<Element<Direction>> createDirections(UUID id, List<Element<DirectionResponse>> responses) {
+            return Lists.newArrayList(ElementUtils.element(id, Direction.builder()
+                .responses(responses)
+                .build()));
+        }
+
+        private List<Element<DirectionResponse>> createDirectionResponses(UUID responseId, UUID directionId) {
+            return Lists.newArrayList(ElementUtils.element(responseId, DirectionResponse.builder()
+                .directionId(directionId)
+                .build()));
+        }
+    }
+
+    //    @Nested
```

Comment:
I know this PR is still WIP, just marking commented code in case you missed it.

Reply:
had forgotten about this, good spot 👍